lib/src/edwards25519/compressed_edwards_y.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:solana/src/edwards25519/edwards_point.dart';
+import 'package:solana/src/edwards25519/extensions.dart';
+import 'package:solana/src/edwards25519/field_element.dart';
+
+final _d = FieldElement([
+  -10913610,
+  13857413,
+  -15372611,
+  6949391,
+  114729,
+  -8787816,
+  -6275908,
+  -3247719,
+  -18696448,
+  -12055116,
+]);
+
+class CompressedEdwardsY {
+  CompressedEdwardsY(this.data);
+
+  final List<int> data;
+
+  EdwardsPoint decompress() {
+    final FieldElement y = FieldElement.fromByteArray(data);
+    final FieldElement ySquare = y.square();
+    final FieldElement u = ySquare.subtract(FieldElement.one);
+    final FieldElement v = ySquare.multiply(_d).add(FieldElement.one);
+    final SqrtRatioM1Result sqrt = FieldElement.sqrtRatioM1(u, v);
+    if (sqrt.wasSquare != 1) {
+      throw Exception('not a valid point');
```

Comment:
`FormatException`

Reply:
It's not a matter of format actually. It's a matter of math correctness. What's the appropriate exception?